runelite-client/src/main/java/net/runelite/client/ui/components/ThinProgressBar.java

```diff
@@ -0,0 +1,76 @@
+package net.runelite.client.ui.components;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import javax.swing.JPanel;
+
+/**
+ * A progress bar to be displayed underneath the GE offer item panels
+ */
+public class ThinProgressBar extends JPanel
+{
+	private int maximumValue;
+	private int value;
+
+	private JPanel topBar;
+
+	public ThinProgressBar()
+	{
+		setLayout(new BorderLayout());
+		setBackground(Color.GREEN.darker());
+
+		topBar = new JPanel();
+		topBar.setPreferredSize(new Dimension(100, 4));
+		topBar.setBackground(Color.GREEN);
+
+		add(topBar, BorderLayout.WEST);
+	}
+
+	public void setMaximumValue(int maximumValue)
+	{
+		this.maximumValue = maximumValue;
+	}
+
+	public void setValue(int value)
+	{
+		this.value = value;
+	}
+
+	/**
+	 * Updates the UI based on the percentage progress
+	 */
+	public void update()
+	{
+		double percentage = getPercentage();
+		int topWidth = (int) (getSize().width * (percentage / 100));
+
+		topBar.setPreferredSize(new Dimension(topWidth, 4));
+		topBar.repaint();
+
+		revalidate();
+		repaint();
+	}
+
+	public double getPercentage()
+	{
+		if (value == 0)
```

Comment:
Why is this if statement necessary? `(0 * 100) / x` is still 0.

Reply:
That's a weird test then. That means no test that accounts for `value = 1`, and `maximumValue = 0`, which would throw an error. I'm guessing there is no test because there is no way for `value` to be greater than zero if the `maximumValue` is zero... Either way, it would be better if the check was `if (maximumValue == 0)` since that would make more sense imo.